Makefile

```
@@ -47,8 +47,37 @@ cover:
 coverview: cover
 	xdg-open _build/test/cover/index.html
 
+# Generate and patch documentation.
+# The patching is needed to be able to generate documentation via Elixirs mix.
+# Following changes are needed:
+# - Handle link targets in headers, changes:
+#     '### <a name="link">Header</a> ###' to
+#     '<a name="link"></a> ### Header ###'
+# - Newline needed for before following tags:
+#     </table> </dd> </pre>
+# - Removal of unneeded line breaks (visual only)
+#
+# Note: sed on macOS requires explicit in-place extensions (-i <extension>)
 edoc:
 	@$(REBAR) edoc
+	@for file in doc/*.md ; do \
+		sed -i.bak 's|### <a name="\(.*\)">\(.*\)</a> ###|<a name="\1"></a>\n### \2 ###|g' $${file} ; \
+		sed -i.bak 's|</table>|\n</table>|g' $${file} ; \
+		sed -i.bak 's|</dd>|\n</dd>|g' $${file} ; \
+		sed -i.bak 's|</code></pre>|</code>\n</pre>|g' $${file} ; \
+		sed -i.bak 's|<br />||g' $${file} ; \
+		rm $${file}.bak ; \
+	done
```

If any of these fails, it will go unnoticed. In a typical make target, you'd normally stop on the first failure. Not sure if it's necessary. We may see if it fails anyway, correct it and publish the corrected docs, but I think it's worth mentioning.

Another thing: I haven't tried this so there may be a typo, but think it's possible to do all replacements in one sed invocation and define the sed script in a variable:

```suggestion
	@for file in doc/*.md ; do \
		sed -i.bak '$(md_replacements)' $${file} && \
		rm $${file}.bak ; \
	done

-define md_replacements
s|### <a name="\(.*\)">\(.*\)</a> ###|<a name="\1"></a>\n### \2 ###|g;
s|</table>|\n</table>|g;
s|</dd>|\n</dd>|g;
s|</code></pre>|</code>\n</pre>|g;
s|<br />||g
-endef

```

---

Interesting.
Sed seemed to be a bit different when running on Mac though, at least regarding the inplace handling.
Maybe @alexandremcosta could test if that works on mac as well

---

Yes, the -i flag differs, but it's fairly new. Sed itself is very old though and you've always been able to have multiple commands in a sed script. Just not exactly sure when it comes to newlines in the makefile and such.

---

Hey guys, I confirm that both versions work on OSX and generate identical .md files

---

Kept inital variant, which correlated with the `eredis` project. Can change if there is a string feeling for it though.